### PR TITLE
[24.0 RELEASE] Changelog update and couple fixes

### PR DESCRIPTION
**Pull request by ridz1208**
### Suggested release notes
```
1. get the 24.0.0 source code
2. Run the following scripts:
    1. `php tools/configuration_check.php `
         _This script is used to validate the configuration of LORIS. It checks configuration settings and server paths to make sure that the application and underlying architecture are set up to allow LORIS to run properly._
    2. `php tools/single_use/SaveUserIDToInstrumentData.php confirm`
         _This script is intended for a one-time use only to restore the value of the `UserID` column of instrument tables and the `UserID` key of the instrument JSON `Data` in the flag table._
    3. `php tools/single_use/remove_publication_users_edit_perm_rel_duplicates.php`
         _This script deletes duplicate rows in publication_users_edit_perm_rel to make sure the patch '2020-01-07-publication_users_edit_perm_rel_pk.sql' will execute successfully_
    4. `php tools/single_use/Cleanup_Special_Chars_Document_Repository.php`
         _This script is written to clean up the files with special characters from the document repository data table as well as clean up the quotes appearing as %22 in the file names in the file system_
    5. `php tools/single_use/Cleanup_Special_Chars_Media.php`
         _This script is written to clean up the files with special characters from the media data table as well as clean up the quotes appearing as %22 in the file names in the file system_
    6. `php tools/single_use/Remove_duplicate_examiners.php`
         _This script is written to clean up duplicate userIDs in the examiners table_
3. Make sure the `RegistrationProjectID` column in the `candidate` table of the dataabse does not contain any NULL values. If it does, either assign an existing project to the candidate/scanner or create a new project for these entries before running the release patch (see "Notes for existing projects" for more details).
4. Source the `SQL/Release_patches/23.0_To_24.0_upgrade.sql` SQL file into your database.
5. Run the following scripts:
    1. `php tools/single_use/populate_visits.php`
        _This script populates the `visits` table based on the config.xml, `session` table and `Visit_Windows` table._
    2. `php tools/single_use/Set_Required_elements_completed_flag.php confirm`
        _This script is written to migrate the `Data_entry_completion_status` field from the instrument table (or Data column of the flag table) to it's own column in the flag table called `Required_elements_completed`._
6. Run `make all` on the LORIS root directory
7. OPTIONAL: Source SQL patches from the `SQL/Archive/24.0/Cleanup` directory at your discretion to remove superfluous or deprecated entries in the database.
8. OPTIONAL: Run the `tools/single_use/Remove_Data_entry_completion_status_instr_column.php` PHP script to generate an SQL patch containing all  `DROP COLUMN` statements for the now unused `Data_entry_completion_status` field in instruments. This script will generate the SQL patch and store it in the `SQL/Archive/autogenerated/single_use/` directory under the name `Remove_Data_entry_completion_status.sql`.
9. OPTIONAL: Remove the `<visitLabel subprojectID="X">` tags in the Config.xml file

***NOTE:** The `tools/single_use/SaveUserIDToInstrumentData.php` can take several hours up to a day to run depending on the size and age of the databse and number of instruments used*
```

### Gaps to fill
1. Does the user need to update any dependencies (PHP, apache, ...)
2.

### CODE REVIEW

File: tools/updateDerivativesTest.php [hunk 1]
```diff
@@ -1,171 +0,0 @@
-#!/usr/bin/php
```
Comment (ridz1208) [inline tools/updateDerivativesTest.php:1]: @driusan deleting this as it is deemed unnecessary by the PR that is creating it (read conversation in https://github.com/aces/Loris/pull/7433)

@AlexandraLivadas please confirm

**Comment by CamilleBeau** [comment]
@ridz1208 Thanks Rida! Will keep you updated

File: tools/updateDerivativesTest.php (hunk 1 above)
Comment (AlexandraLivadas) [reply tools/updateDerivativesTest.php:1]: @ridz1208 Yes this is unnecessary and can be deleted!

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
-- A PSR3 compatible logging interface was added (PR #7509)
-- Added support for Amazon S3 (PR #7963)
-
 #### Features
 - Data tables may now stream data as they're loading rather than waiting
   until all data has loaded. (PR #6853)
-- Login module adds toggle visibility for password field.
+- Added support for Amazon S3 (PR #7963)
+- Added ability for scripts to bulk load instrument data (PR #6869)
+- New data dictionary framework and module (#6936)
```
Comment (cmadjar) [inline CHANGELOG.md:18]:
```suggestion
- New data dictionary framework and module (PR #6936)
```

Comment (cmadjar) [reply CHANGELOG.md:18]: (just for consistency)

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+- New classes to describe a data dictionary (PR #6938)
 
 #### Updates and Improvements
 - Module-specific permissions added for Survey Accounts, Imaging Behavioural
 Quality Control, and Behavioural Quality Control. (PR #6041)
 - Addition of a new `account_request_date` in `users` table that will be used when
 requesting a new account and will be displayed in the User Accounts module (PR #6191)
 - Candidate's age can be retrieved from the Candidate class in days, months, or years (PR #5945)
-- Addition of autoSelect prop to React SelectElement allows for auto-selection of only available select option (PR #6156)
+- Addition of autoSelect prop to React SelectElement allows for auto-selection of
```
Comment (cmadjar) [inline CHANGELOG.md:27]:
```suggestion
- Addition of `autoSelect` property to React `SelectElement` allows for auto-selection of 
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
-- The default value of the ScannerID field of the mri_protocol table is now NULL instead of 0 (PR #7496).
+- Field `Data_entry_completion_status` given its own column in flag, and renamed to 
+`Required_elements_completed` (PR #6876)
+- Addition of `changeProject` and `resetProject` helper functions to the 
+`LorisIntegrationTest` class to help test project permissions (PR #6912)
+- Unit tests added for the `NDB_BVL_Instrument`, `NDB_Page`, `NDB_Factory`, 
+`User`, and `UserPermissions` libraries (PR #6819, PR #6804, PR #6776, PR #6765)
+- The name of a Standard Date field in a LINST instrument must now end with the 
+string `_date`. Otherwise, a LORIS exception is thrown. There is no restriction 
+on the naming format of a Basic Date or MonthYear field. (PR #6923)
```
Comment (cmadjar) [inline CHANGELOG.md:38]:
```suggestion
on the naming format of a Basic Date or `MonthYear` field. (PR #6923) 
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
-- Unit tests added for the `NDB_BVL_Instrument`, `NDB_Page`, `NDB_Factory`, `User`, and `UserPermissions` libraries (PR #6819, PR #6804, PR #6776, PR #6765)
-- The name of a Standard Date field in a LINST instrument must now end with the string `_date`. Otherwise, a LORIS exception is thrown. There is no restriction on the naming format of a Basic Date or MonthYear field. (PR #6923) 
-- React Form Select Element now has the ability to set an option in the element as a disabled option. (PR #7306)
 - Addition of `date_format` as a DataType in ConfigSettings (PR #6719)
-- New Data Dictionary framework to better describe data (PR #6936)
 - Addition of new tables to store PET HRRT data (PR #6142)
 - Modification of the `parameter_file` table's `Value` field type to `longtext` (PR #7392)
 - Addition of 4 configuration settings for the minc2bids converter (PR #7488)
+- A PSR3 compatible logging interface was added (PR #7509)
+- Multiple classes of errors flagged by phan are now fixed (various PRs)
```
Comment (cmadjar) [inline CHANGELOG.md:48]:
```suggestion
- Multiple classes of errors flagged by `phan` are now fixed (multiple PRs)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+- LorisInstance class was added to represent an installed LORIS instance (PR #6118)
```
Comment (cmadjar) [inline CHANGELOG.md:49]:
```suggestion
- A `LorisInstance` class was added to represent an installed LORIS instance (PR #6118)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+- Validation for DateElement (JS). (PR #7266)
```
Comment (cmadjar) [inline CHANGELOG.md:50]:
```suggestion
- Validation for `DateElement` (JS). (PR #7266)
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+- Session Current_stage default value changed for Not Started (PR #7102)
```
Comment (cmadjar) [inline CHANGELOG.md:51]:
```suggestion
- Session `Current_stage` default value changed for `Not Started` (PR #7102)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+- Permissions were reorganized and categorized in the user_accounts module (PR #7327)
```
Comment (cmadjar) [inline CHANGELOG.md:52]:
```suggestion
- Permissions were reorganized and categorized in the user accounts module (PR #7327)
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
-- A LINST instrument Date field name now appears correctly (not truncated) on the instrument if it includes the string `_date`. (PR #6923)
-- A subtest with only static and/or static score fields (i.e. no values to submit) can be saved with no errors so that the instrument scoring script can be called. (PR #7124)
-- When a superuser edits another user, the labels for each permission is correctly displayed (PR #7451) 
+- A LINST instrument Date field name now appears correctly (not truncated) on the 
+instrument if it includes the string `_date`. (PR #6923)
+- A subtest with only static and/or static score fields (i.e. no values to submit) 
+can be saved with no errors so that the instrument scoring script can be called. (PR #7124)
+- When a superuser edits another user, the labels for each permission is correctly 
+displayed (PR #7451) 
+- The imaging uploader now starts automatically if you have the imaging uploader
```
Comment (cmadjar) [inline CHANGELOG.md:67]:
```suggestion
- The imaging insertion pipeline (LORIS-MRI scripts) now starts automatically if you have the imaging uploader 
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
-- A subtest with only static and/or static score fields (i.e. no values to submit) can be saved with no errors so that the instrument scoring script can be called. (PR #7124)
-- When a superuser edits another user, the labels for each permission is correctly displayed (PR #7451) 
+- A LINST instrument Date field name now appears correctly (not truncated) on the 
+instrument if it includes the string `_date`. (PR #6923)
+- A subtest with only static and/or static score fields (i.e. no values to submit) 
+can be saved with no errors so that the instrument scoring script can be called. (PR #7124)
+- When a superuser edits another user, the labels for each permission is correctly 
+displayed (PR #7451) 
+- The imaging uploader now starts automatically if you have the imaging uploader 
+auto-launch set to true and your current upload overwrites an existing file (PR #7084).
```
Comment (cmadjar) [inline CHANGELOG.md:68]:
```suggestion
auto-launch set to `true` and your current upload overwrites an existing file (PR #7084).
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+- Script CouchDB_MRI_Importer now computes the correct names the data dictionary
```
Comment (cmadjar) [inline CHANGELOG.md:69]:
```suggestion
- Script `CouchDB_MRI_Importer` now computes the correct names in the data dictionary 
```

Comment (cmadjar) [reply CHANGELOG.md:69]: in or from? not sure. Feels like it is missing something between `names` and `the`

File: CHANGELOG.md [hunk 13]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+- New integration test class to test project permissions (PR #6912)
 #### Configuration
 - Addition of configuration settings for the DICOM to BIDS insertion pipeline (PR #7937)
 - Addition of configuration settings for the MINC to BIDS converter script (PR #7488)
-#### Candidate Profile
-- New integration test class to test project permissions (PR #6912)
+#### Conflict Resolver
+- Changes are now saved automatically, one by one. Once a conflict is resolved
+the cell that contains the input field will glow green. It is possible to change
+the resolved conflicts to a new value until the page is refreshed. [(PR #7558)](https://github.com/aces/Loris/pull/7558)
```
Comment (cmadjar) [inline CHANGELOG.md:108]:
```suggestion
the resolved conflicts to a new value until the page is refreshed. (PR #7558)
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+#### Instruments
+- General help text added for instrument data entry (PR #6902)
 - Fix to avoid select with required option in group fields to display as multiselect (PR #7254)
-- Fixes to insert JSON intrument (PR #7155)
-#### Create timepoint
-- Fix a reindexing of the languages array which caused a database insert error. (PR #7145)
+- Fixes to insert JSON instruments (PR #7155)
+#### Issue Tracker
+- Readability of comments and history was improved. (PR #6138)
+- Update validation to allow NULL Site (For All Sites issues) (#6526)
```
Comment (cmadjar) [inline CHANGELOG.md:143]:
```suggestion
- Update validation to allow `NULL` Site (For All Sites issues) (#6526)
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+- Removal for the need of the `VisitLabel` section of the `config.xml` file. All 
+visit configurations and their association to projects are now in the database (#7663 & #7729)
 - Removal of references to Reliability module in Raisinbread (PR #6895)
 - Raisinbread visit stage inconsistency changed (PR #6896)
 - HRRT patch sourced to Raisinbread (PR #6897)
 - Improved consistency of RB data: field UploadLocation of table mri_upload now has an appropriate value (PR #7086)
+- The default value of the `ScannerID` field of the `mri_protocol` table is now 
+`NULL` instead of `0`. This means that if a protocol is valid on all the study's 
+scanners, then `ScannerID` of the protocol should be set to `NULL` (PR #7496)
+
```
Comment (cmadjar) [inline CHANGELOG.md:187]: Should the new cleanup MySQL patches be mentioned here? Or maybe under Notes for existing projects...

File: CHANGELOG.md [hunk 16]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
-- Deletion of `monthYearFields` variable in instrument class. all references to this variable should be removed from project instruments.
-- Visit definitions is no longer done in the `config.xml`. An importer tool is available to import the current setup into the `visit` table of the database. Make sure the visits displayed after these changes are what you expect. (#7663 & #7729)
-- There is a new abstract "getDataDictionary" function in the instrument class for the new data dictionary framework. This is already implemented for LINST instruments, and existing instruments using LorisForm can use the `\LorisFormDictionaryImpl` trait to extract it in the same way as `lorisform_parser.php` did for the old datadict module.
-- The name of a Standard Date field in a LINST instrument must now end with the string `_date`. Otherwise, a LORIS exception is thrown. There is no restriction on the naming format of a Basic Date or MonthYear field. (PR #6923) 
+- New function `Candidate::getSubjectForMostRecentVisit` replaces `Utility::getSubprojectIDUsingCandID`, 
+adding ability to determine which subproject a candidate belongs to given their most recent visit.
```
Comment (cmadjar) [inline CHANGELOG.md:191]:
```suggestion
adding the ability to determine which subproject a candidate belongs to given their most recent visit.
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+- LINST instrument class was modified to implement the `getFullName()` and `getSubtestList()`
+functions thus making entries in the test_names and instrument_subtests tables
```
Comment (cmadjar) [inline CHANGELOG.md:193]:
```suggestion
functions thus making entries in the `test_names` and `instrument_subtests` tables 
```

File: CHANGELOG.md [hunk 18]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
-- The name of a Standard Date field in a LINST instrument must now end with the string `_date`. Otherwise, a LORIS exception is thrown. There is no restriction on the naming format of a Basic Date or MonthYear field. (PR #6923) 
+- New function `Candidate::getSubjectForMostRecentVisit` replaces `Utility::getSubprojectIDUsingCandID`, 
+adding ability to determine which subproject a candidate belongs to given their most recent visit.
+- LINST instrument class was modified to implement the `getFullName()` and `getSubtestList()`
+functions thus making entries in the test_names and instrument_subtests tables 
+unnecessary for LINST instruments (PR #7169)
+- The script `tools/single_use/remove_publication_users_edit_perm_rel_duplicates.php`
+should be run before upgrading LORIS. 
+- The `Data_entry_completion_status` column of instrument tables has been migrated 
+to its own column in flag, and renamed to `Required_elements_completed`. After script
```
Comment (cmadjar) [inline CHANGELOG.md:198]:
```suggestion
to its own column in the `flag` table, and renamed to `Required_elements_completed`. After script 
```

File: CHANGELOG.md [hunk 19]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+`updateDataEntryCompletionStatus` are called in any overrides, make sure to replace 
+all instances with their newly named counterparts, `_setRequiredElementsCompletedFlag`, 
+`_determineRequiredElementsCompletedFlag`, `updateRequiredElementsCompletedFlag`. (PR #6876)
+- Deprecation of `begintable` and `endtable` elements in LINST instruments (PR #7183)
+- Deletion of `dateTimeFields` and `monthYearFields` variables in instrument class. 
+All references to these variables should be removed from project instruments. (PR #7183)
+- Visit definitions is no longer done in the `config.xml`. An importer tool (`populate_visits.php`) is 
+available to import the current setup into the `visit` table of the database. 
+Make sure the visits displayed after these changes are what you expect. (#7663 & #7729)
+- There is a new abstract "getDataDictionary" function in the instrument class
```
Comment (cmadjar) [inline CHANGELOG.md:213]:
```suggestion
- There is a new abstract `getDataDictionary` function in the instrument class 
```

File: CHANGELOG.md [hunk 20]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+`_determineRequiredElementsCompletedFlag`, `updateRequiredElementsCompletedFlag`. (PR #6876)
+- Deprecation of `begintable` and `endtable` elements in LINST instruments (PR #7183)
+- Deletion of `dateTimeFields` and `monthYearFields` variables in instrument class. 
+All references to these variables should be removed from project instruments. (PR #7183)
+- Visit definitions is no longer done in the `config.xml`. An importer tool (`populate_visits.php`) is 
+available to import the current setup into the `visit` table of the database. 
+Make sure the visits displayed after these changes are what you expect. (#7663 & #7729)
+- There is a new abstract "getDataDictionary" function in the instrument class 
+for the new data dictionary framework. This is already implemented for LINST instruments, 
+and existing instruments using LorisForm can use the `\LorisFormDictionaryImpl`
```
Comment (cmadjar) [inline CHANGELOG.md:215]:
```suggestion
and existing instruments using `LorisForm` can use the `\LorisFormDictionaryImpl` 
```

File: CHANGELOG.md [hunk 21]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+- Deprecation of `begintable` and `endtable` elements in LINST instruments (PR #7183)
+- Deletion of `dateTimeFields` and `monthYearFields` variables in instrument class. 
+All references to these variables should be removed from project instruments. (PR #7183)
+- Visit definitions is no longer done in the `config.xml`. An importer tool (`populate_visits.php`) is 
+available to import the current setup into the `visit` table of the database. 
+Make sure the visits displayed after these changes are what you expect. (#7663 & #7729)
+- There is a new abstract "getDataDictionary" function in the instrument class 
+for the new data dictionary framework. This is already implemented for LINST instruments, 
+and existing instruments using LorisForm can use the `\LorisFormDictionaryImpl` 
+trait to extract it in the same way as `lorisform_parser.php` did for the old datadict module.
```
Comment (cmadjar) [inline CHANGELOG.md:216]:
```suggestion
trait to extract it in the same way as `lorisform_parser.php` did for the old `datadict` module.
```

**Review by cmadjar — CHANGES_REQUESTED** [review]
Wow! That is impressive. So much work!!

@ridz1208 I made a few edit suggestions. Most of them are very minor. Thank you for putting everything together!!

File: CHANGELOG.md (hunk 15 above)
Comment (ridz1208) [reply CHANGELOG.md:187]: both is better than none... can you make sure to add it if it's missing (I dont really have the context)

File: CHANGELOG.md [hunk 22]
```diff
@@ -10,171 +10,218 @@ changes in the following format: PR #1234***
...
+for the new data dictionary framework. This is already implemented for LINST instruments, 
+and existing instruments using `LorisForm` can use the `\LorisFormDictionaryImpl` 
+trait to extract it in the same way as `lorisform_parser.php` did for the old `datadict` module.
+- The name of a Standard Date field in a LINST instrument must now end with the 
+string `_date`. Otherwise, a LORIS exception is thrown. There is no restriction 
+on the naming format of a Basic Date or MonthYear field. (PR #6923) 
+- Run `tools/single_use/Remove_duplicate_examiners.php` to remove duplicate 
+examiners that may have been created before bugfix. Make sure to run this script 
+_before_ running the `SQL/New_patches/2021-09-28-Unique_examiners.sql`. (PR #7462)
+- Deletion of support for the oldest version of the API (v0.0.2) (PR #6944)
```
Comment (cmadjar) [inline CHANGELOG.md:223]:
```suggestion
- Deletion of support for the oldest version of the API (v0.0.2) (PR #6944)
- Refer to instructions for clean up scripts meant to remove unused data in the `SQL/Archive/24.0/Cleanup` directory
```

**Review by cmadjar — APPROVED** [review]
LGTM. @driusan ready for you

File: CHANGELOG.md [hunk 23]
```diff
@@ -10,171 +10,219 @@ changes in the following format: PR #1234***
...
 ### Clean Up
-- Removal of unused variables and unnecessary branching from `getBattery()` and `getBatteryVerbose()` functions (PR #7167)
-- Removal of the violated_scans_edit permission (PR #6747)
-- Removal for the need of the `VisitLabel` section of the `config.xml` file. All Visit configurations and their association to projects are now in the database (#7663 & #7729)
-- Removal of unused variables and unnecessary branching from `getBattery()` and `getBatteryVerbose()` functions (PR #7167)
+- Removal of unused variables and unnecessary branching from `getBattery()` and 
+`getBatteryVerbose()` functions (PR #7167)
+- Removal of the `violated_scans_edit` permission (PR #6747)
+- Removal for the need of the `VisitLabel` section of the `config.xml` file. All 
+visit configurations and their association to projects are now in the database (#7663 & #7729)
```
Comment (CamilleBeau) [inline CHANGELOG.md:179]:
```suggestion
- Removal for the need of the `VisitLabel` section of the `config.xml` file. All 
visit configurations and their association to projects are now in the database.
The visits can be removed from the `config.xml` file after running the 
`populate_visits.php` script only. (#7663 & #7729)
```

File: CHANGELOG.md [hunk 24]
```diff
@@ -10,171 +10,219 @@ changes in the following format: PR #1234***
...
+- If `_setDataEntryCompletionStatus`, `_determineDataEntryCompletionStatus`, and/or 
+`updateDataEntryCompletionStatus` are called in any overrides, make sure to replace 
+all instances with their newly named counterparts, `_setRequiredElementsCompletedFlag`, 
+`_determineRequiredElementsCompletedFlag`, `updateRequiredElementsCompletedFlag`. (PR #6876)
+- Deprecation of `begintable` and `endtable` elements in LINST instruments (PR #7183)
+- Deletion of `dateTimeFields` and `monthYearFields` variables in instrument class. 
+All references to these variables should be removed from project instruments. (PR #7183)
+- Visit definitions is no longer done in the `config.xml`. An importer tool (`populate_visits.php`) is 
+available to import the current setup into the `visit` table of the database. 
+Make sure the visits displayed after these changes are what you expect. (#7663 & #7729)
```
Comment (CamilleBeau) [inline CHANGELOG.md:212]:
```suggestion
- Visit definitions is no longer done in the `config.xml`. An importer tool (`populate_visits.php`) is 
available to import the current setup into the `visit` table of the database. 
Make sure the visits displayed after these changes are what you expect. 
The visits can be removed from the `config.xml` file after the importer tool is run. (#7663 & #7729)
```

**Review by CamilleBeau — DISMISSED** [review]

**Pull request merged**